### PR TITLE
TYP: add return types to read_sql|read_sql_query|read_sql_table

### PR DESCRIPTION
**Pull request by topper-123**
Adds return types to read_sql|read_sql_query|read_sql_table.

I'd like to take on the other pd.read_* functions in follow-ups.

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -170,8 +172,49 @@ def read_sql_table(
...
+    index_col=None,
+    coerce_float=True,
+    parse_dates=None,
+    columns=None,
+    chunksize: int = 1,
+) -> Iterator[DataFrame]:
+    ...
+
+
+@overload
```
Comment (WillAyd) [inline pandas/io/sql.py:194]: Hmm I don't think an overload should match the overloaded-signature should it? I think can delete?

**Review by WillAyd — COMMENTED** [review]
looks pretty good. I think the second overload in each case is unnecessary though

@simonjayhawkins

File: pandas/io/sql.py (hunk 1 above)
Comment (topper-123) [reply pandas/io/sql.py]: Yeah, that's right. I've updated the PR to remove all unneeded overloads.

File: pandas/io/sql.py [hunk 2]
```diff
@@ -257,15 +286,41 @@ def read_sql_table(
...
+def read_sql_query(
+    sql,
+    con,
+    index_col=None,
+    coerce_float=True,
+    params=None,
+    parse_dates=None,
+    chunksize: None = None,
+) -> Iterator[DataFrame]:
```
Comment (simonjayhawkins) [inline pandas/io/sql.py:311]: are the return types of the two overloads switched? or have a misread the docstring?

Comment (WillAyd) [reply pandas/io/sql.py:311]: Yea looks like a copy / paste error? The other two funcs look correct.

lgtm outside of this issue

**Comment by topper-123** [comment]
Yeah, that was a mistake. I've updated.

**Comment by simonjayhawkins** [comment]
Thanks @topper-123

**Pull request merged**